blackbox_repository/conversion_scripts/scripts/nasbench201_import.py

```diff
@@ -0,0 +1,216 @@
+    latency = np.repeat(np.expand_dims(latency, axis=-1), n_fidelities, axis=-1)
+    save_objective_values_helper('latency', latency)
+
+    flops = np.array([data['arch2infos'][ai]['200']['all_results'][(dataset, 777)]['flop'] for ai in range(n_hps)])
+    flops = np.repeat(np.expand_dims(flops, axis=-1), n_seeds, axis=-1)
+    flops = np.repeat(np.expand_dims(flops, axis=-1), n_fidelities, axis=-1)
+    save_objective_values_helper('flops', flops)
+
+    params = np.array([data['arch2infos'][ai]['200']['all_results'][(dataset, 777)]['params'] for ai in range(n_hps)])
+    params = np.repeat(np.expand_dims(params, axis=-1), n_seeds, axis=-1)
+    params = np.repeat(np.expand_dims(params, axis=-1), n_fidelities, axis=-1)
+    save_objective_values_helper('params', params)
+
+    configuration_space = {
+        node: search_space.choice(['avg_pool_3x3', 'nor_conv_3x3', 'skip_connect', 'nor_conv_1x1', 'none'])
+        for node in hp_cols
+    }
+
+    fidelity_space = {
+        "hp_epoch": search_space.randint(lower=1, upper=201)
+    }
+
+    return BlackboxTabular(
+        hyperparameters=hyperparameters,
+        configuration_space=configuration_space,
+        fidelity_space=fidelity_space,
+        objectives_evaluations=objective_evaluations,
+        fidelity_values=fidelity_values,
+        objectives_names=[f"metric_{m}" for m in objective_names],
+    )
+
+
+def generate_nasbench201():
+    blackbox_name = "nasbench201"
+    file_name = repository_path / "NATS-tss-v1_0-3ffb9.pickle.pbz2"
+    if not file_name.exists():
+        print(f"did not find {file_name}, downloading")
+        import requests
+
+        def download_file_from_google_drive(id, destination):
+            URL = "https://docs.google.com/uc?export=download"
+
+            session = requests.Session()
+
+            response = session.get(URL, params={'id': id}, stream=True)
+            token = get_confirm_token(response)
+
+            if token:
+                params = {'id': id, 'confirm': token}
+                response = session.get(URL, params=params, stream=True)
+
+            save_response_content(response, destination)
+
+        def get_confirm_token(response):
+            for key, value in response.cookies.items():
+                if key.startswith('download_warning'):
+                    return value
+
+            return None
+
+        def save_response_content(response, destination):
+            CHUNK_SIZE = 32768
+
+            with open(destination, "wb") as f:
+                for chunk in response.iter_content(CHUNK_SIZE):
+                    if chunk:  # filter out keep-alive new chunks
+                        f.write(chunk)
+
+        download_file_from_google_drive('1vzyK0UVH2D3fTpa1_dSWnp1gvGpAxRul', file_name)
+
+    f = bz2.BZ2File(file_name, 'rb')
+    data = pickle.load(f)
+
+    with catchtime("converting"):
+        bb_dict = {}
+        for dataset in ['cifar10', 'cifar100', 'ImageNet16-120'][:1]:
```

Comment:
'cifar10' was 'cifar10-valid' before. This was always annoying, I am happy to change it. But are you sure this works here? In the current code by David, there is no explicit mentioning of the dataset names, so I suppose it is still 'cifar10-valid' there.

Reply:
I agree cifar10 is easier. 

Not sure I got your other comment Matthias, the name of the dataset appears in the mapping in bb_dict in the repository.

Reply:
So it is still cifar10-valid ? What I meant is that Aaron's code has these names explicitly in the code, but in David's code I do not find them, so I suppose they come from the file that is loaded?

Reply:
AFAIK there are two different version of cifar10 in nb201, and uses the validation error as optimization criterion and on e the test error. To be consistent with the other two datasets cifar100 and Imagenet-16, I switched now to the version based on the test error. Unfortunately for none of the dataset we have another hold out test error as we have for  FCNet based NAS-HPO-Bench